### PR TITLE
update nix to 0.9.0

### PR DESCRIPTION
**Pull request by andy128k**
At least it compiles and tests are passed :)
Related to https://github.com/xd009642/tarpaulin/issues/45

### CODE REVIEW

File: src/ptrace_control.rs [hunk 1]
```diff
@@ -2,58 +2,59 @@ use std::ptr;
...
     ptrace_setoptions(pid, options)
 }
 
-pub fn detach_child(pid: pid_t) -> Result<c_long> {
+pub fn detach_child(pid: Pid) -> Result<c_long> {
     ptrace(PTRACE_DETACH, pid, ptr::null_mut(), ptr::null_mut())
 }
 
-pub fn continue_exec(pid: pid_t, sig: Option<Signal>) -> Result<c_long> {
+pub fn continue_exec(pid: Pid, sig: Option<Signal>) -> Result<c_long> {
```
Comment (xd009642) [inline src/ptrace_control.rs]: So all instances of continue_exec can be replaced by `sys::ptrace::cont` in nix. NIx has a bunch of new utility functions that fulfil a lot of what ptrace_control did for me

File: src/ptrace_control.rs [hunk 2]
```diff
@@ -2,58 +2,59 @@ use std::ptr;
...
-pub fn trace_children(pid: pid_t) -> Result<()> {
+pub fn trace_children(pid: Pid) -> Result<()> {
     //TODO need to check support.
     let options: PtraceOptions = PTRACE_O_TRACESYSGOOD | PTRACE_O_TRACEEXEC | PTRACE_O_TRACEEXIT |
         PTRACE_O_TRACECLONE | PTRACE_O_TRACEFORK | PTRACE_O_TRACEVFORK;
     ptrace_setoptions(pid, options)
 }
 
-pub fn detach_child(pid: pid_t) -> Result<c_long> {
+pub fn detach_child(pid: Pid) -> Result<c_long> {
```
Comment (xd009642) [inline src/ptrace_control.rs]: Can be replaced with sys::ptrace::detach

File: src/ptrace_control.rs [hunk 3]
```diff
@@ -2,58 +2,59 @@ use std::ptr;
...
     ptrace(PTRACE_POKEUSER, pid, RIP as * mut c_void, pc as * mut c_void)
 }
 
 pub fn request_trace() -> Result<c_long> {
-    ptrace(PTRACE_TRACEME, 0, ptr::null_mut(), ptr::null_mut())
+    ptrace(PTRACE_TRACEME, Pid::from_raw(0), ptr::null_mut(), ptr::null_mut())
 }
 
-pub fn get_event_data(pid: pid_t) -> Result<c_long> {
+pub fn get_event_data(pid: Pid) -> Result<c_long> {
```
Comment (xd009642) [inline src/ptrace_control.rs]: Can be replaced with `sys::ptrace::getevent`

File: src/ptrace_control.rs [hunk 4]
```diff
@@ -2,58 +2,59 @@ use std::ptr;
...
+pub fn current_instruction_pointer(pid: Pid) -> Result<c_long> {
     ptrace(PTRACE_PEEKUSER, pid, RIP as * mut c_void, ptr::null_mut())
 }
 
-pub fn set_instruction_pointer(pid: pid_t, pc: u64) -> Result<c_long> {
+pub fn set_instruction_pointer(pid: Pid, pc: u64) -> Result<c_long> {
     ptrace(PTRACE_POKEUSER, pid, RIP as * mut c_void, pc as * mut c_void)
 }
 
 pub fn request_trace() -> Result<c_long> {
```
Comment (xd009642) [inline src/ptrace_control.rs]: Can be replaced by `sys::ptrace::traceme`

**Comment by xd009642** [comment]
Thanks for this, just some small changes again but all pretty small in comparison to the work you've done :smile: Thanks again!

**Comment by andy128k** [comment]
Functions, you've mentioned, are not released yet.

**Comment by xd009642** [comment]
Ah my bad, in that case merged!

**Pull request merged**